### PR TITLE
Factor out non-Material parts of MaterialApp into WidgetsApp

### LINKED ISSUE #1346: Separate out generic application wiring from MaterialApp
Create another class, like App or BasicApp, that does things like wire up locale and media query information, but does not assume any particular UI layout the way MaterialApp does.

MaterialApp could then either inherit from this class, or hold one in a member variable, depending on which seems cleaner.

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/1346

### CODE REVIEW

File: packages/flutter/lib/src/widgets/app.dart [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    assert(showSemanticsDebugger != null);
+  }
+
+  /// A one-line description of this app for use in the window manager.
+  final String title;
+
+  /// The default text style for [Text] in the application.
+  final TextStyle textStyle;
+
+  /// The accent color to use for the application in the operating system
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/app.dart]: accent  <--- not sure if this term makes sense here.  Maybe just drop this word.

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
Maybe call it `BasicApp` ?  We could have a whole suit of "basic" widgets that don't involve material.

**Comment by Hixie** [comment]
The problem with "BasicApp" is it looks weird when you're making an app with this widget ("my app isn't a basic app!"). I agree that "WidgetsApp" isn't particularly great either though.

File: packages/flutter/lib/src/widgets/app.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/app.dart]: Changed to "primary color" which seems to be what the Android docs call it.

**Comment by abarth** [comment]
I guess `App` is another possibility.  Dunno.  Whatever you think is best.

**Comment by Hixie** [comment]
I think if we use App, we should use that at the Material layer.

I went with WidgetsApp. I am very open to renaming it further though.

**Comment by krisgiesing** [comment]
Maybe FlutterApp? Anyway, thanks for the refactoring :)

**Comment by Hixie** [comment]
FlutterApp could work...

**Pull request merged**